controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1493,6 +1513,21 @@ func (r *BareMetalHostReconciler) handleDataImageActions(prov provisioner.Provis
 		return actionContinue{}
 	}
 
+	bmhProvState := info.host.Status.Provisioning.State
+	isBMHAvailable := bmhProvState == metal3api.StateAvailable
```

Comment:
nit: I'd prefer this function to have an argument like `detachOnly bool` and keep the provision state logic at the caller side (where you already check for the Available state).

Reply:
Makes sense, I will make the changes.